source/kubernetes/versions.rst

```diff
@@ -0,0 +1,116 @@
++===============+==================================================================+
+| MAJOR         | versions that may make incompatible API changes                  |
++---------------+------------------------------------------------------------------+
+| MINOR         | versions that adds functionality in a backwards compatible manner|
++---------------+------------------------------------------------------------------+
+| PATCH         | versions that makes backwards compatible bug fixes               |
++---------------+------------------------------------------------------------------+
+
+For example:
+
+.. code-block:: bash
+
+  [major].[minor].[patch]
+  
+  v1.18.2
+  v1.17.5
+  v1.16.9
+
+Catalyst Cloud Kubernetes Service uses cluster template to manage each Kubernetes
+version and the matrix of addons running on top of Kubernetes cluster. And users
+should be able to see the Kubernetes version from the cluster template name. For
+example: *kubernetes-v1.16.9-prod-20200602*
+
+
+For more information, see `Kubernetes Release Versioning`_.
+
+.. _`Semantic Versioning`: http://semver.org/
+.. _`Kubernetes Release Versioning`: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#kubernetes-release-versioning
+
+Kubernetes Versions Support Policy
+==================================
+
+Catalyst Cloud Kubernetes Service supports at least ``3`` minors versions. As long
+as there is a new minor version released, Catalyst Cloud Kubernetes Service will
+try to get it certified (pass the CNCF conformance test) and released in ``30``
+days. And then deprecate the oldest minor version. If the current 3 minor versions 
+are v1.17.x, v1.16.x and v1.15.x. Then as long as the new v1.18.x released, the
+v1.15.x versions(both the two patch versions) will be removed and out of supported.
+Here, out of support means whenever users ask for support, you will be asked
+to upgrade your clusters to supported versions first.
+
+Catalyst Cloud Kubernetes Service supports the latest ``2`` stable patch versions
+for each version. As long as there is a patch version released, the oldest patch
+version will be removed and out of supported. For example, if current versions
+supported for v1.16.x are v1.16.9 and v1.16.8, then v1.16.8 will be removed in
+favor of the release of v1.16.10.
+
+.. code-block:: bash
+
+    v1.18.2, v1.18.1, v1.17.5, v1.17.4, v1.16.9, v1.16.8
+
+Users should always aim to run the latest patch version for each minor version
+to get the latest security enhancements. For example, if the current Kubernetes
+cluster is running on v1.16.9 and the new patch version is v1.16.10, then it
+is highly recommended to upgrade to v1.16.10 as soon as possible.
+
+When there is a new patch version released, users have ``30`` days to upgrade
+to a supported patch version if cluster's current version is deprecated by the
+new version released.
```

Comment:
How should the user know when a template they're using is deprecated? What pages or channels should they use to be notified, can we link to them from here?

Reply:
Good questions, @evhan 

Before having notification support on Horizon, we may have to use the release notes and this versions page to indicate the currently supported versions.

Whenever there is a new version template releaed or deprecated, it will be published on the release note page.
Meanwhile, a table on this versions page will show all the current supported versions and a warning note that the version is going to be deprecated. Does that make sense for you?

Reply:
That makes sense. It does mean users will have to be proactive, and I expect many will not notice the deprecations if it requires polling this page. But, I think that's OK as long as that's explained here.

I think it was also suggested somewhere that there be some kind of warning on the clusters page as well? That would also be very helpful, since users are more likely to look at that page than this one.

Reply:
Good point, I will think about how to leverage the cluster page to show some useful information.